src/adapter.ts

```diff
@@ -12,155 +12,184 @@
+
+  // authenticate is the destructor for Adapter.
+  private async authenticate(): Promise<void> {
+    await this.sequelize.authenticate();
+  }
+
+  // sync is the destructor for Adapter.
+  private async sync(): Promise<void> {
+    await this.casbinRule.sync();
+  }
+
+  private loadPolicyLine(line: any, model: Model): void {
+    let lineText = line.p_type;
+    if (line.v0) {
+      lineText += ', ' + line.v0;
     }
-
-    private open() {
-        this.sequelize = new Sequelize(this.connStr);
-
-        this.createTable();
+    if (line.v1) {
+      lineText += ', ' + line.v1;
     }
-
-    private close() {
-        this.sequelize.close();
+    if (line.v2) {
+      lineText += ', ' + line.v2;
     }
-
-    private createTable() {
+    if (line.v3) {
+      lineText += ', ' + line.v3;
     }
-
-    private dropTable() {
+    if (line.v4) {
+      lineText += ', ' + line.v4;
     }
-
-    private loadPolicyLine(line: {[index: string]: string}, model: Model) {
-        let lineText = line.ptype;
-        if (line.v0 !== '') {
-            lineText += ', ' + line.v0;
-        }
-        if (line.v1 !== '') {
-            lineText += ', ' + line.v1;
-        }
-        if (line.v2 !== '') {
-            lineText += ', ' + line.v2;
-        }
-        if (line.v3 !== '') {
-            lineText += ', ' + line.v3;
-        }
-        if (line.v4 !== '') {
-            lineText += ', ' + line.v4;
-        }
-        if (line.v5 !== '') {
-            lineText += ', ' + line.v5;
-        }
-
-        Helper.loadPolicyLine(lineText, model);
+    if (line.v5) {
+      lineText += ', ' + line.v5;
     }
+    Helper.loadPolicyLine(lineText, model);
+  }
 
-    /**
-     * loadPolicy loads all policy rules from the storage.
-     */
-    public loadPolicy(model: Model) {
-        const lines = CasbinRule.findAll();
+  // loadPolicy loads policy from database.
+  public async loadPolicy(model: Model) {
+    const {casbinRule} = this;
+    const lines = await casbinRule.findAll();
 
-        for (const line of lines) {
-            this.loadPolicyLine(line.toJSON(), model);
-        }
+    for (const line of lines) {
+      this.loadPolicyLine(line.toJSON(), model);
     }
+  }
 
-    private savePolicyLine(ptype: string, rule: string[]): {[index: string]: string} {
-        const line = new CasbinRule();
-
-        line.ptype = ptype;
-        if (rule.length > 0) {
-            line.v0 = rule[0];
-        }
-        if (rule.length > 1) {
-            line.v1 = rule[1];
-        }
-        if (rule.length > 2) {
-            line.v2 = rule[2];
-        }
-        if (rule.length > 3) {
-            line.v3 = rule[3];
-        }
-        if (rule.length > 4) {
-            line.v4 = rule[4];
-        }
-        if (rule.length > 5) {
-            line.v5 = rule[5];
-        }
+  private savePolicyLine(ptype: string, rules: string[]): CasbinRuleAttributes {
+    const line: any = {p_type: ptype};
 
-        return line;
+    if (rules.length > 0) {
```

Comment:
Do not just copy your code here. I have fixed your typo ``rules`` to ``rule``, and you changed it back again.

Reply:
Why the variable must be singular? But it's plural at here.

Reply:
In Casbin, a string array like [ptype, v0, v1, v2] is one policy rule. So it's singular.